### ISSUE TITLE
[no-unsafe-assignment] doesn't handle interface implementation

### ISSUE THREAD

**Issue by danielnixon** [danielnixon/eslint-plugin-total-functions#40]
From https://github.com/microsoft/TypeScript/issues/18770#issuecomment-612634795

```ts
type I = {
  readonly a: string;
};

// eslint-disable-next-line functional/no-class
class C implements I {
  // This should be flagged by no-unsafe-assignment or another (new) rule
  // eslint-disable-next-line functional/prefer-readonly-type
  a = "";
}

const D = new C();

// eslint-disable-next-line functional/no-expression-statement, functional/immutable-data
D.a = "something";
```

**Comment by danielnixon** [comment]
Similarly:

```ts
interface Shape {
  readonly color: string;
}

interface Square extends Shape {
  color: string;
}
```

**Comment by danielnixon** [comment]
And:

```ts

type Shape = {
  readonly color: string;
};

type Square = Shape & {
  // eslint-disable-next-line functional/prefer-readonly-type
  color: string;
};
```
